gh-87286: Add a number of LOG_* constants to syslog

ronaldoussoren opened the pull request:
This adds a number of syslog facilities to the syslogmodule.c.
These values are available on macOS.


<!-- gh-issue-number: gh-87286 -->
* Issue: gh-87286
<!-- /gh-issue-number -->

ronaldoussoren committed:
bpo-43120: Add a number of LOG_* constants to syslog

This adds a number of syslog facilities to the syslogmodule.c.
These values are available on macOS.

akulakov commented on Doc/library/syslog.rst:
```diff
@@ -90,7 +90,12 @@ Facilities:
+   :const:`LOG_REMOTEAUTH`, :const:`LOG_INSTALL`, :const:`LOG_RAS`.
+
+   .. versionchanged:: 3.10
+       Added :const:`LOG_NETINFO`, :const:`LOG_REMOTEAUTH`, :const:`LOG_INSTALL`,
```
also LOG_FTP

akulakov commented on Misc/NEWS.d/next/Library/2021-02-03-21-26-57.bpo-43120.4fbCBl.rst:
```diff
@@ -0,0 +1,2 @@
+Added :const:`LOG_NETINFO`, :const:`LOG_REMOTEAUTH`, :const:`LOG_INSTALL`,
```
also LOG_FTP

akulakov commented on Doc/library/syslog.rst:
```diff
@@ -90,7 +90,12 @@ Facilities:
+   :const:`LOG_AUTHPRIV`, :const:`LOG_FTP`, :const:`LOG_NETINFO`, 
+   :const:`LOG_REMOTEAUTH`, :const:`LOG_INSTALL`, :const:`LOG_RAS`.
+
+   .. versionchanged:: 3.10
```
Should this be 3.11 now?

akulakov reviewed (commented):
By the way I also was not able to find some of these levels in `man 3 syslog` on my MacOS, (for example LOG_NETINFO, LOG_RAS, etc - but it's old, 10.12, so were they added more recently? I couldn't find them online either.

ronaldoussoren committed:
Merge branch 'main' into bpo-43120

ronaldoussoren committed:
Minor updates to bring this PR back to live

ronaldoussoren committed:
Merge branch 'bpo-43120' of github.com:ronaldoussoren/cpython into bpo-43120

ronaldoussoren committed:
Fixes

* Fix patchcheck errors
* Rename NEWS.d entry to the right name (issue# istead of PR#)

ronaldoussoren committed:
Switch contant documentation to the data directive

This fixes a CI warning and matches the pattern
used in the documentation for ``os``.

hugovk commented on Doc/library/syslog.rst:
```diff
@@ -107,22 +107,62 @@ The module defines the following functions:
+          LOG_INFO
+          LOG_DEBUG
+
+   Priority levels (high to low):
```
This applies to the preceding list:
```suggestion
   Priority levels (high to low).
```

serhiy-storchaka commented on Doc/library/syslog.rst:
```diff
@@ -107,22 +107,62 @@ The module defines the following functions:
+          LOG_INFO
+          LOG_DEBUG
+
+   Priority levels (high to low):
```
Should not it be above the previous list?

hugovk replied:
Could be, currently it renders like this:

![image](https://github.com/python/cpython/assets/1324225/e0478f51-85d7-4e32-b014-ecd45268bd48)

Where this text is the ["description details"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dd) of the last "description term":

![image](https://github.com/python/cpython/assets/1324225/32bb1862-827a-4dcd-8245-feec8884460c)

hugovk commented on Doc/library/syslog.rst:
```diff
@@ -107,22 +107,62 @@ The module defines the following functions:
+          LOG_AUTHPRIV
+          LOG_CRON
+          LOG_DAEMON
+          LOG_FTP 
```
Lint nit:
```suggestion
          LOG_FTP
```

ronaldoussoren committed:
Update Doc/library/syslog.rst

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

ronaldoussoren committed:
Update Doc/library/syslog.rst

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

ronaldoussoren committed:
Fix last sphinx markup warning in syslog documetation

AlexWaygood commented on Doc/library/syslog.rst:
```diff
@@ -107,22 +107,62 @@ The module defines the following functions:
+          LOG_LOCAL6
+          LOG_LOCAL7
+
+   Facilities, depending on availabiltiy in ``<syslog.h>`` for :const:`LOG_AUTHPRIV`,
```
Typo :)

```suggestion
   Facilities, depending on availability in ``<syslog.h>`` for :const:`LOG_AUTHPRIV`,
```

ronaldoussoren replied:
Spelling is hard. Thanks!

ronaldoussoren committed:
Update Doc/library/syslog.rst

Co-authored-by: Alex Waygood <Alex.Waygood@Gmail.com>

terryjreedy commented on Misc/NEWS.d/next/Library/2023-12-07-16-55-41.gh-issue-87286.MILC9_.rst:
```diff
@@ -0,0 +1,3 @@
+Added :const:`LOG_FTP`, :const:`LOG_NETINFO`, :const:`LOG_REMOTEAUTH`,
+:const:`LOG_INSTALL`, :const:`LOG_RAS`, and :const:`LOG_LAUNCHD` tot the
+:mod:`syslog` module, all of them constants on used on macOS.
```
I believe 'tot' should be 'to' and 'constants on used on' should be 'constants used on'.
